src/gen/conditions.rs

```diff
@@ -3665,3 +3691,79 @@ fn test_relative_condition_on_ephemeral(
         }
     }
 }
+
+#[cfg(test)]
+#[rstest]
+// the scale factor is 10000
+#[case("((2 (1 )", 10000)]
+// 0 is OK (but does it make sense?)
+#[case("((2 (0 )", 0)]
+// the cost accumulates
+#[case("((2 (1 ) ((2 (2 ) ((2 (3 )", (1 + 2 + 3) * 10000)]
```

Comment:
What's the format for these conditions? It seems weird that the parens don't balance.

Reply:
these tests use a simple parser where `(` means `0xff` and `)` means `0x80`, so all open-parentheses just take two items and are implicitly terminated